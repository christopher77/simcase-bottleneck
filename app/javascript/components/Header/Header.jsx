import React from "react";
import styles from "./Header.module.css";
import universityLogo from "../../assets/Onboarding/universityLogo.svg";
import { navigate } from "@reach/router";
import Modal from "../ui/Modal";
import alert from "../../assets/alert.svg";

const Header = props => {
  const { feedback, rightHeader, reset, resetScores } = props;
  const [showModal, setShowModal] = React.useState(false);

  function changeRightHeader(value) {
    switch (parseInt(value)) {
      case 1:
        return "First Quarter";
        break;
      case 2:
        return "Second Quarter";
        break;
      case 3:
        return "Third Quarter";
        break;
      case 4:
        return "Fourth Quarter";
        break;
      case 5:
        return "Score";
        break;
      case 6:
        return "Feedback";
        break;
      case 7:
        return "Leaderboard";
        break;
      case 8:
        return "Onboarding";
        break;
      case 9:
        return "Leaderboard";
        break;
    }
  }

  const openModal = () => {
    setShowModal(true);
  };

  const closeModal = () => {
    setShowModal(false);
  };

  const goHome = () => {
    reset();
    resetScores();
    navigate("/");
  };

  return (
    <>
      <div className={styles.headerContainer}>
        <div
          className={styles.homebutton}
          onClick={parseInt(rightHeader) > 6 ? goHome : openModal}
        >
          <img src={universityLogo} className={styles.logo} />
        </div>
        <h3 className={rightHeader == 7 ? styles.titleWhite : styles.title}>
          {feedback ? changeRightHeader(6) : changeRightHeader(rightHeader)}
        </h3>
      </div>

      {showModal && (
        <Modal>
          <h5 className={styles.messageModal}>
            You will lose all points, Are you sure?{" "}
          </h5>
          <img src={alert} alt="alert-image"></img>
          <button className={styles.buttonConfirm} onClick={goHome}>
            Continue
          </button>
          <button className={styles.buttonCancel} onClick={closeModal}>
            Cancel
          </button>
        </Modal>
      )}
    </>
  );
};

export default Header;
