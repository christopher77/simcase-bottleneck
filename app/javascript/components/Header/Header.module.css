.headerContainer {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.logo {
  margin: 15px 20px 3px;
}

.title {
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  margin: 20px 20px 3px;
}
.titleWhite {
  composes: title;
  color: white;
}

.buttonConfirm {
  background: #73c056;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 47px;
  color: black;
  width: 10em;
  height: 42px;
  font-size: 14px;
  padding: 8px 12px;
  text-transform: uppercase;
  font-size: 17px;
  font-weight: bold;
  border-color: transparent;
  cursor: pointer;
  margin-top: 10px;
  margin-left: initial;
}
.buttonCancel {
  composes: buttonConfirm;
  background: white;
  color: #4f4f4f;
  box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, 0.25);
  margin-top: 15px;
}
.buttonConfirm:active {
  transform: scale(1.1);
}
.buttonCancel:active {
  transform: scale(1.1);
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}
.homebutton {
  cursor: pointer;
}
.messageModal {
  font-size: 0.8rem;
  font-weight: bold;
  line-height: 1.1rem;
  padding-bottom: 10px;
  text-align: center;
  width: 100%;
}
