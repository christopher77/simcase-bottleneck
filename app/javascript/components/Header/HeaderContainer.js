import { connect } from "react-redux";
import Header from "./Header";
import { reset } from "../../redux/actions/RoundsActions";
import { resetScores } from "../../redux/actions/UserActions";

const mapStateToProps = state => {
  return {
    feedback: state.feedback.feedback
  };
};

const mapDispatchToProps = {
  reset: reset,
  resetScores: resetScores
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
