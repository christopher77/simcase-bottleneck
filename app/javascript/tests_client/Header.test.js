import React from "react";
import { render } from "@testing-library/react";
import Header from "../components/Header/Header";

test("Header component", () => {
  const { asFragment } = render(<Header />);
  expect(asFragment()).toMatchSnapshot();
});
