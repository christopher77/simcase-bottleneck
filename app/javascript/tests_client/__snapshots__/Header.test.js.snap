// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component 1`] = `
<DocumentFragment>
  <div
    class="headerContainer"
  >
    <div
      class="homebutton"
    >
      <img
        class="logo"
        src="[object Object]"
      />
    </div>
    <h3
      class="title"
    >
      Onboarding
    </h3>
  </div>
</DocumentFragment>
`;
